submarine-server/server-submitter/submitter-k8s/src/main/java/org/apache/submarine/server/submitter/k8s/K8sJobSubmitter.java

```diff
@@ -186,6 +197,80 @@ private Job parseResponseObject(Object object, ParseOp op) throws SubmarineRunti
     throw new SubmarineRuntimeException(500, "K8s Submitter parse upstream response failed.");
   }
 
+  public String getLog(final Job job) {
+    if (job == null) {
+      return null;
+    }
+    final CoreV1Api COREV1_API = new CoreV1Api(client);
+
+    try {
+      final V1PodList podList = COREV1_API.listNamespacedPod(
+          job.getSpec().getSubmitterSpec().getNamespace(),
+          "false", null, null,
+          getJobLabelSelector(job), null, null,
+          null, null);
+
+      final V1Pod pod = podList.getItems().get(0);
+      final String podName = pod.getMetadata().getName();
+      final String namespace = job.getSpec().getSubmitterSpec().getNamespace();
+      
+      String readNamespacedPodLog =
+          COREV1_API.readNamespacedPodLog(
+            podName,
+            namespace,
+            null,
+            Boolean.FALSE,
+            Integer.MAX_VALUE,
+            null,
+            Boolean.FALSE,
+            Integer.MAX_VALUE,
+            40,
+            Boolean.FALSE);
+
+      return readNamespacedPodLog;
+
+    } catch (final ApiException e) {
+      LOG.warn("Error when listing pod for job:" + job.toString(), e.getMessage());
+    }
+    return null;
+  }
+
+  public InputStream getLogStream(final Job job) {
+    if (job == null) {
+      return null;
+    }
+    final CoreV1Api COREV1_API = new CoreV1Api(client);
+
+    try {
+      final V1PodList podList = COREV1_API.listNamespacedPod(
+          job.getSpec().getSubmitterSpec().getNamespace(),
+          "false", null, null,
+          getJobLabelSelector(job), null, null,
+          null, null);
+      final V1Pod pod = podList.getItems().get(0);
```

Comment:
Also, is it possible that we merge different streams into one? And update the aggregated log stream?

Reply:
I am not sure how to render the mixed stream in the browser ...
Maybe something like this?
[
{"pod1": "log start"},
{"pod1": "1%"},
{"pod2": "log start"},
{"pod1": "2%"},
{"pod2": "1%"},
....